packages/trader/src/App/Components/Elements/market-countdown-timer.jsx

```diff
@@ -0,0 +1,140 @@
+    const date = moment(date_target).format('YYYY-MM-DD');
+    const api_res = await getTradingTimes(date);
+    if (!api_res.api_initial_load_error) {
+        const { times } = getSymbol(target_symbol, api_res.trading_times);
+        const { open, close } = times;
+        const is_closed_all_day = open.length === 1 && open[0] === '--' && close[0] === '--';
+        if (is_closed_all_day) {
+            // check tomorrow trading times
+            await whenMarketOpens(days_offset + 1, target_symbol);
+        } else {
+            const date_str = date_target.toISOString().substring(0, 11);
+            const getUTCDate = hour => new Date(`${date_str}${hour}Z`);
+            for (let i = 0; i < open.length; i++) {
+                const diff = +getUTCDate(open[i]) - +date_target;
+                if (diff > 0) {
+                    when_market_opens = +getUTCDate(open[i]);
+                    break;
+                }
+            }
+        }
+    }
+    return when_market_opens;
+};
+
+const calculateTimeLeft = when_market_opens => {
+    const difference = when_market_opens - +new Date();
+    let timeLeft = {};
+
+    if (difference > 0) {
+        timeLeft = {
+            days: Math.floor(difference / (1000 * 60 * 60 * 24)),
+            hours: Math.floor((difference / (1000 * 60 * 60)) % 24),
+            minutes: Math.floor((difference / 1000 / 60) % 60),
+            seconds: Math.floor((difference / 1000) % 60),
+        };
+    }
+
+    return timeLeft;
+};
+
+const padWithZero = number => {
+    if (number < 10) {
+        return `0${number}`;
+    }
+    return number;
+};
+
+const MarketCountdownTimer = ({ symbol }) => {
+    const [when_market_opens, setWhenMarketOpens] = React.useState('');
+    const [time_left, setTimeLeft] = React.useState(calculateTimeLeft(when_market_opens));
+
+    React.useEffect(() => {
+        async function fetchTradingTimes() {
+            const response = await whenMarketOpens(0, symbol);
+            setWhenMarketOpens(response);
+        }
+        fetchTradingTimes();
+    }, []);
+
+    React.useEffect(() => {
+        setTimeout(() => {
+            setTimeLeft(calculateTimeLeft(when_market_opens));
+        }, 1000);
+    });
+
+    let timer_components = '';
+
+    Object.keys(time_left).forEach(interval => {
+        if (interval === 'days') {
+            if (time_left[interval]) {
+                timer_components += `${time_left[interval]} ${interval} `;
+            }
+        } else {
+            const value = time_left[interval];
+            timer_components += interval !== 'seconds' ? `${padWithZero(value)}:` : `${padWithZero(value)}`;
+        }
+    });
+
+    let view = null;
+    if (when_market_opens) {
+        view = (
+            <React.Fragment>
+                <p>
+                    <Localize i18n_default_text='Reopens:' />
+                </p>
+                <p className='market-is-closed-overlay__timer'>{timer_components} GMT</p>
```

Comment:
Shouldn't we translate `GMT`?

Reply:
Checked our code base, we don't do this.